app/controllers/application_controller.rb

```diff
@@ -1,28 +1,16 @@
 class ApplicationController < ActionController::Base
-
   protect_from_forgery with: :exception
 
-  helper_method :current_user,
-                :logged_in?
-
-  private
-
-  def authenticate_user!
-    unless current_user
-      cookies[:path] = request.path
+  before_action :configure_permitted_parameters, if: :devise_controller?
 
-      return redirect_to login_path, alert: 'Are you a Guru? Verify your Email and Password please.'
-    end
-
-    cookies[:email] = current_user.email
-  end
+  protected
 
-  def current_user
-    @current_user ||= User.find_by(id: session[:user_id]) if session[:user_id]
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up, keys: %i[first_name last_name])
   end
 
-  def logged_in?
-    current_user.present?
+  def after_sign_in_path_for(user)
+    current_user.admin? ? admin_tests_path : tests_path
```

Comment:
```suggestion
    user.admin? ? admin_tests_path : tests_path
```
это метод девайса, он принимает аргумент
более правильно следовать его структуре и использовать именно аргумент, приходящий в метод, а не вызывать хелпер `current_user`